src/libYARP_OS/include/yarp/os/Timer.h

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2017 iCub Facility
+ * Authors: Andrea Ruzzenenti
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include <yarp/os/RateThread.h>
+#include <yarp/os/Mutex.h>
+#include <functional>
+
+namespace yarp {
+    namespace os {
+        class Timer;
+        struct YarpTimerEvent;
+    }
+}
+
+struct yarp::os::YarpTimerEvent
+{   
+    /**
+     * @brief lastExpected when the last callback actually happened
+     */
+    double lastExpected;
+
+    /**
+     * @brief lastReal when the last callback actually happened
+     */
+    double lastReal;
+
+    /**
+     * @brief currentExpected this is when the current callback should have been called
+     */
+    double currentExpected;
+
+    /**
+     * @brief currentReal When the current callback is actually being called
+     */
+    double currentReal;
+
+    /**
+     * @brief lastDuration Contains the duration of the last callback
+     */
+    double lastDuration;
+
+    /**
+     * @brief runCount the count of calls
+     */
+    unsigned int runCount;
+};
+
+class YARP_OS_API yarp::os::Timer : public yarp::os::RateThread {
+    typedef std::function<void(const yarp::os::YarpTimerEvent&)> TimerCallback;
```

Comment:
This `typedef` should probably be public, since it is used in public methods...

Maybe it could be useful if the callback would return a bool = false when the Timer should be stopped?